File: de.carsten_lenz.liquibasegui/src/de/carsten_lenz/liquibasegui/util/StringToFileConverter.scala
package de.carsten_lenz.liquibasegui.util

import org.eclipse.core.databinding.conversion._
import java.io.File

class StringToFileConverter extends Converter(classOf[String], classOf[File]) {
    def convert(o: java.lang.Object) = o match {
        case s: String if !s.isEmpty => new File(s)
        case _ => null
    }
}

File: de.carsten_lenz.liquibasegui/src/de/carsten_lenz/liquibasegui/util/StringToFileConverter.java
package de.carsten_lenz.liquibasegui.util;

import java.io.File;

import org.eclipse.core.databinding.conversion.Converter;

public class StringToFileConverter extends Converter {

    public StringToFileConverter() {
        super(String.class, File.class);
    }

    @Override
    public Object convert(Object fromObject) {
        if (fromObject instanceof String) {
            return new File((String) fromObject);
        }
        return null;
    }
    
}

